tools/check_volume_levels.m

```diff
@@ -0,0 +1,251 @@
+function check_volume_levels(cmd, fn1, fn2, fn3)
+
+% check_volume_levels(cmd, fn1, fn2, fn3)
+%
+% Inputs
+%   cmd - Use 'generate' or 'analyze'
+%   fn1 - File name for sine wave to generate or first record file name to analyze
+%   fn2 - File name to analyze 2nd
+%   fn3 - File name to analyze 3rd
+%
+% E.g.
+%   check_volume_levels('generate','sine.wav');
+%   check_volume_levels('measure','rec1.wav','rec2.wav','rec3.wav');
+
+% SPDX-License-Identifier: BSD-3-Clause
+% Copyright(c) 2016 Intel Corporation. All rights reserved.
+% Author: Seppo Ingalsuo <seppo.ingalsuo@linux.intel.com>
+
+	addpath('../../sof/tools/test/audio/std_utils');
+	addpath('../../sof/tools/test/audio/test_utils');
+
+	if exist('OCTAVE_VERSION', 'builtin')
+		pkg load signal
+	end
+
+	switch lower(cmd)
+		case 'generate'
+			pass = generate(fn1);
+			if pass
+				fprintf(1, 'done\n');
+			else
+				fprintf(1, 'FAIL\n');
+				exit(1);
+			end
+		case 'measure'
+			pass = measure(fn1, fn2, fn3);
+			if pass
+				fprintf(1, 'PASS\n');
+			else
+				fprintf(1, 'FAIL\n');
+				exit(1);
```

Comment:
shall we remove those 2 'exit(1)' as they are making my octave hang when running it with gui mode, logging errors should be enough?

Reply:
This script is not for GUI mode but for batch processing. For interactive tests with GUI I've commented out those lines. The GUI doesn't hang but it quits with exit().

Reply:
I'm not aware of simple ways in script to check if we are in Octave GUI. If such exist we could make a custom exit function.

Reply:
>  This script is not for GUI mode but for batch processing. 

It would be great to find a solution that supports both but waiting for that should IMHO not block this "automation" PR.

Reply:
@singalsu @marc-hb just removing this 'exit(1)' line will work for both GUI and batch processing per my testing.

Reply:
The shell script thinks a failed test passed if I omit the exit(1).

Reply:
Okay, got you now @singalsu

Reply:
> The shell script thinks a failed test passed if I omit the exit(1).

Thanks @singalsu for "testing the test code" and making sure it can report failures. I found this to be incredibly rare in this project.

After only 2 minutes googling (sorry) I found "error()", did you try it? https://octave.org/doc/v4.0.1/Raising-Errors.html